cpp/src/transform/transform.cpp

```diff
@@ -89,6 +167,24 @@ std::unique_ptr<column> transform(column_view const& input,
   return output;
 }
 
+std::unique_ptr<column> generalized_masked_op(table_view data_view,
+                                              std::string const& udf,
+                                              data_type output_type,
+                                              rmm::mr::device_memory_resource* mr)
+{
+  rmm::cuda_stream_view stream   = rmm::cuda_stream_default;
+  std::unique_ptr<column> output = make_fixed_width_column(output_type, data_view.num_rows());
+  std::unique_ptr<column> output_mask =
+    make_fixed_width_column(cudf::data_type{cudf::type_id::BOOL8}, data_view.num_rows());
```

Comment:
As a future optimization, maybe we can further improve the null_mask handling by creating the bit arrays directly with:
https://github.com/rapidsai/cudf/blob/0a4e8a1c860518c9e30f5289dea07bd0879fb642/cpp/include/cudf/null_mask.hpp#L84-L87

And then, inside `generalized_operation` you can set the corresponding bits using
https://github.com/rapidsai/cudf/blob/0a4e8a1c860518c9e30f5289dea07bd0879fb642/cpp/include/cudf/utilities/bit.hpp#L83
(unsafe set should be fine since you are just doing row-wise op)

Probably saves the boolean array casting.

Reply:
From offline: since this is at thread level operation, we can't use unsafe set here. Using the safe set version results many atomics and becomes a trade off of performance and memory usage.